shell.nix

```diff
@@ -0,0 +1,47 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+pkgs.mkShell rec {
+  buildInputs = with pkgs; [
+    rustup
+    cargo-nextest cargo-fuzz
+    typos
+    fd yq
+
+    gdb lldb rr
+    clang llvmPackages.libclang
+    libgcc.lib
+    openssl pkg-config cmake
+    evcxr tokei
+    libxkbcommon
+    wayland xorg.libX11 xorg.libXcursor xorg.libXrandr xorg.libXi
+    alsa-lib
+    fontconfig freetype
+
+    shaderc directx-shader-compiler
+    libGL
+    vulkan-headers vulkan-loader
+    vulkan-tools vulkan-tools-lunarg
+    vulkan-validation-layers
+    vulkan-extension-layer
+
+    python3 valgrind
+    renderdoc
+    gnuplot
```

Comment:
it'd probably be good to document which of these are manditory and which are nice-to-haves. And why we need them.